compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -566,9 +566,30 @@ bool compareIntrinsics(const GPUMatmulShapeType &problem,
     return lhsArea > rhsArea;
   }
 
-  // Finally if everything else is the same, prefer large K size.
-  return ShapedType::getNumElements(lhs.kSizes) >
-         ShapedType::getNumElements(rhs.kSizes);
+  // Prefer large K size here.
+  int64_t lhsKSize = ShapedType::getNumElements(lhs.kSizes);
+  int64_t rhsKSize = ShapedType::getNumElements(rhs.kSizes);
+  if (lhsKSize != rhsKSize) {
+    return lhsKSize > rhsKSize;
+  }
```

Comment:
An easier way to write this comparison is to come up with a feature tuple for `GPUInstrinsicType` and then delegate the exact comparison logic to the `std::tuple` comparison operator.